### PR TITLE
Implements drop_na()

### PR DESCRIPTION
**Pull request by markfairbanks**

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -28,7 +28,11 @@
   data tables).
 
 * We have begun the process of add translations for tidyr verbs beginning
-  with `pivot_wider()` (@markfairbanks, #189).
+  with:
+  
+  * `drop_na()` (@markfairbanks, #194)
```
Comment (hadley) [inline NEWS.md:33]: dtplyr just made it to CRAN so once you merge/rebase, you'll need to move this to a new bullet in the dev version.

File: R/step-call.R [hunk 2]
```diff
@@ -231,3 +231,43 @@ unique.dtplyr_step <- function(x, incomparables = FALSE, ...) {
...
+  step_call(data, "na.omit", args = args)
+}
+
+# exported onLoad
+drop_na.data.table <- function(data, ...) {
+  data <- lazy_dt(data)
+  drop_na(data, ...)
+}
+
+globalVariables("drop_na")
```
Comment (hadley) [inline R/step-call.R:273]: You can instead add to `dt_funs` in `tidyeval.R`

File: tests/testthat/test-step-call.R [hunk 3]
```diff
@@ -161,3 +161,69 @@ test_that("unique is an alias for distinct", {
...
+
+# drop_na ------------------------------------------------------------------
+
+test_that("empty call drops every row", {
+  df <- lazy_dt(tibble(x = c(1, 2, NA), y = c("a", NA, "b")), "DT")
+  exp <- tibble(x = c(1), y = c("a"))
+  step <- drop_na(df)
+  res <- as_tibble(step)
+  expect_equal(show_query(step), expr(na.omit(DT)))
+  expect_equal(res, exp)
```
Comment (hadley) [inline tests/testthat/test-step-call.R:173]: I _think_ this is a little easier to follow:

```suggestion
  tb <- tibble(x = c(1, 2, NA), y = c("a", NA, "b")
  dt <- drop_na(lazy_dt(tb, "DT"))
  expect_equal(show_query(dt), expr(na.omit(DT)))
  expect_equal(as_tibble(step), tb[1, ])
```

File: tests/testthat/test-step-call.R [hunk 4]
```diff
@@ -161,3 +161,69 @@ test_that("unique is an alias for distinct", {
...
+  step <- drop_na(df, x)
+  res <- as_tibble(step)
+  expect_equal(show_query(step), expr(na.omit(DT, cols = "x")))
+  expect_equal(res, exp)
+  exp <- tibble(x = c(1), y = c("a"))
+  res <- as_tibble(drop_na(df, x:y))
+  expect_equal(res, exp)
+})
+
+test_that("groups are preserved", {
```
Comment (hadley) [inline tests/testthat/test-step-call.R:193]: Is it necessary to test this? It's `step_call()` that takes care of this, not your code.

File: tests/testthat/test-step-call.R [hunk 5]
```diff
@@ -161,3 +161,69 @@ test_that("unique is an alias for distinct", {
...
+
+  gdf <- group_by(df, "g")
+  gexp <- dplyr::group_by(exp, "g")
+
+  res <- collect(drop_na(gdf, y))
+  expect_equal(res, gexp)
+  expect_equal(dplyr::group_vars(res), dplyr::group_vars(gexp))
+})
+
+test_that("empty call drops every row", {
```
Comment (hadley) [inline tests/testthat/test-step-call.R:205]: Duplicate of first test?

File: tests/testthat/test-step-call.R [hunk 6]
```diff
@@ -161,3 +161,69 @@ test_that("unique is an alias for distinct", {
...
+  df <- lazy_dt(tibble(x = c(1, 2, NA), y = c("a", NA, "b")))
+  res <- as_tibble(drop_na(df))
+  expect_identical(res, tibble(x = 1, y = "a"))
+})
+
+test_that("errors are raised", {
+  df <- lazy_dt(tibble(x = c(1, 2, NA), y = c("a", NA, "b")))
+  expect_error(collect(drop_na(df, !! list())))
+  expect_error(collect(drop_na(df, "z")))
```
Comment (hadley) [inline tests/testthat/test-step-call.R:214]: These should be `expect_snapshot(error = TRUE, ...)` so we can see the precise errors

File: tests/testthat/test-step-call.R [hunk 7]
```diff
@@ -161,3 +161,69 @@ test_that("unique is an alias for distinct", {
...
+  exp <- data.frame(x = c(1, 2))
+  expect_equal(res, exp)
+})
+
+test_that("preserves attributes", {
+  df <- lazy_dt(tibble(x = structure(c(1, NA), attr = "!")))
+  rs <- collect(drop_na(df))
+
+  expect_equal(rs$x, structure(1, attr = "!"))
+})
```
Comment (hadley) [inline tests/testthat/test-step-call.R:229]: I think these are data.frame specific so can be dropped.

**Review by hadley — COMMENTED** [review]
Looks good, thanks!

File: R/step-call.R (hunk 2 above)
Comment (markfairbanks) [reply R/step-call.R:273]: Isn't `dt_funs` just for data.table functions?

Comment (hadley) [reply R/step-call.R:273]: Oh sorry, I totally misinterpreted the problem. You should use `tidyr::drop_na()` in the `drop_na` method.

File: tests/testthat/test-step-call.R [hunk 8]
```diff
@@ -161,3 +161,36 @@ test_that("unique is an alias for distinct", {
...
+  expect_equal(res, exp)
+  exp <- tibble(x = c(1), y = c("a"))
+  res <- as_tibble(drop_na(df, x:y))
+  expect_equal(res, exp)
+})
+
+test_that("errors are raised", {
+  tb <- tibble(x = c(1, 2, NA), y = c("a", NA, "b"))
+  dt <- lazy_dt(tb, "DT")
+  expect_snapshot(collect(drop_na(dt, !! list())), error = TRUE)
```
Comment (hadley) [inline tests/testthat/test-step-call.R:194]: I'm not sure what this is supposed to test, and it looks like the `!!` is being evaluated too early. So I suggest just dropping it.

**Comment by markfairbanks** [comment]
All set for review

**Comment by hadley** [comment]
Thanks!

**Pull request merged**